### PR TITLE
Feature/system

### PR DESCRIPTION
**Pull request by shekeraoleksandr**
Implemented all system features.

Also:
- Add boolean method overload at the DataType class with probability parameter 
- Updated Helper class with objectKey and maybe methods
- Add mac address generation  at the Internet class
- Add hexadecimal number generation

### CODE REVIEW

**Comment by shekeraoleksandr** [comment]
I'll solve all errors a little bit later

File: include/faker-cxx/System.h [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+#include <algorithm>
+#include <ctime>
+#include <random>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "../src/common/StringHelper.h"
```
Comment (cieslarmichal) [inline include/faker-cxx/System.h:11]: Please move all needed types in header file to include/types directory, move only types not all the declared variables from those files.

File: src/modules/system/data/CronOptions.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace faker
+{
+struct CronOptions {
```
Comment (cieslarmichal) [inline src/modules/system/data/CronOptions.h:8]: Move struct into include/types

File: src/modules/system/data/FileOptions.h [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+namespace faker
+{
+struct FileOptions {
```
Comment (cieslarmichal) [inline src/modules/system/data/FileOptions.h:5]: Move struct into include/types

File: src/modules/system/data/NetworkInterfaceOptions.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <vector>
+#include <unordered_map>
+
+namespace faker
+{
+struct NetworkInterfaceOptions {
```
Comment (cieslarmichal) [inline src/modules/system/data/NetworkInterfaceOptions.h:10]: Move struct into include/types

File: src/modules/system/data/CronOptions.h [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    bool includeYear = false;
+    bool includeNonStandard = false;
+};
+
+const std::vector<std::string> CRON_DAY_OF_WEEK = {"SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"};
```
Comment (cieslarmichal) [inline src/modules/system/data/CronOptions.h:13]: Please use camelCase for all types of variables

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
Great work! Small changes requested :)

**Comment by shekeraoleksandr** [comment]
Bit busy with other things :( so I'll do it tomorrow

**Comment by cieslarmichal** [comment]
Sure, no problem :)

File: src/modules/system/data/CronOptions.h (hunk 2 above)
Comment (shekeraoleksandr) [reply src/modules/system/data/CronOptions.h]: Done

File: src/modules/system/data/NetworkInterfaceOptions.h (hunk 4 above)
Comment (shekeraoleksandr) [reply src/modules/system/data/NetworkInterfaceOptions.h]: Done

File: src/modules/system/data/CronOptions.h (hunk 5 above)
Comment (shekeraoleksandr) [reply src/modules/system/data/CronOptions.h]: Done

File: include/faker-cxx/System.h [hunk 6]
```diff
@@ -9,17 +9,21 @@
 #include <vector>
 
 #include "../src/common/StringHelper.h"
-#include "../src/modules/system/data/CronOptions.h"
-#include "../src/modules/system/data/FileOptions.h"
-#include "../src/modules/system/data/NetworkInterfaceOptions.h"
-#include "../src/modules/system/data/MimeTypes.h"
 #include "../src/modules/system/data/DirectoryPath.h"
+#include "../src/modules/system/data/MimeTypes.h"
```
Comment (cieslarmichal) [inline include/faker-cxx/System.h:13]: All types included and needed in header file should be in include/types directory as its part of the faker api

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
Please move other needed types from header to include/types directory

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**